gene/database.py

```diff
@@ -159,3 +185,127 @@ def create_meta_data_table(self, existing_tables: List[str]):
                     'WriteCapacityUnits': 10
                 }
             )
+
+    def get_record_by_id(self, concept_id: str,
+                         case_sensitive: bool = True,
+                         merge: bool = False) -> Optional[Dict]:
+        """Fetch record corresponding to provided concept ID
+        :param str concept_id: concept ID for gene record
+        :param bool case_sensitive: if true, performs exact lookup, which is
+            more efficient. Otherwise, performs filter operation, which
+            doesn't require correct casing.
+        :param bool merge: if true, look for merged record; look for identity
+            record otherwise.
+        :return: complete gene record, if match is found; None otherwise
+        """
+        try:
+            if merge:
+                pk = f'{concept_id.lower()}##merger'
+            else:
+                pk = f'{concept_id.lower()}##identity'
+            if case_sensitive:
+                match = self.genes.get_item(Key={
+                    'label_and_type': pk,
+                    'concept_id': concept_id
+                })
+                return match['Item']
+            else:
+                exp = Key('label_and_type').eq(pk)
+                response = self.genes.query(KeyConditionExpression=exp)
+                return response['Items'][0]
+        except ClientError as e:
+            logger.error(f"boto3 client error on get_records_by_id for "
+                         f"search term {concept_id}: "
+                         f"{e.response['Error']['Message']}")
+            return None
+        except (KeyError, IndexError):  # record doesn't exist
+            return None
+
+    def get_records_by_type(self, query: str,
+                            match_type: str) -> List[Dict]:
+        """Retrieve records for given query and match type.
+        :param query: string to match against
+        :param str match_type: type of match to look for. Should be one
+            of {"label", "alias", "xref", "associated_with"} (use
```

Comment:
is "symbol" also an acceptable match type for `get_records_by_type`?

Reply:
Oops. Didn't update docs. It should be symbol, prev_symbol, alias, xref, and associated_with!

Reply:
Oops. Didn't update docs. It should be symbol, prev_symbol, alias, xref, and associated_with!